libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/MapStyleLoader.kt

```diff
@@ -85,6 +86,10 @@ internal class MapStyleLoader(
     }
 
     private val onStyleLoadedListener = OnStyleLoadedListener {
-        _loadedMapStyle.value = mapboxMap?.getStyle()
+        _loadedMapStyle.value = mapboxMap?.getStyle()?.also {
+            // initialize route line layers to ensure they exist before
+            // both RouteLineComponent and RouteArrowComponent start rendering
+            MapboxRouteLineView(options.routeLineOptions.value).initializeLayers(it)
```

Comment:
Why are we creating a new `MapboxRouteLineView` here? Every time the style is loaded, we are creating `RouteLineComponent`
```
reloadOnChange(
    context.mapStyleLoader.loadedMapStyle,
    context.options.routeLineOptions
) { _, lineOptions ->
    routeLineComponent(lineOptions, mapView)
}
```
This `routeLineComponent` creates a new `MapboxRouteLineView` instance. 

Hence, we can just add `initializeLayers` to `RouteLineComponent`.

```
init {
    ....
    ....
    mapboxMap.getStyle()?.apply {
        routeLineView.initializeLayers(this)
    }
}
```
@tomaszrybakiewicz

Reply:
Updated.